### PR TITLE
Go challenge 3

### PR DESCRIPTION
**Pull request by makernaren**

### CODE REVIEW

**Comment by jarydremillard** [comment]
Why do you have two files;
max_occurance.go & max_occurance_test.go?

**Comment by makernaren** [comment]
One is the source code, other is for testing it. I guess thats the standard format for writing tests for golang code. `yourCode_test.go`

File: challenge_3/go/makernaren/max_occurance.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// most number of times.
+package main
+
+import (
+	"fmt"
+)
+
+// Finds the maximum occurance, takes in interface as input and output params
+//as elements in the given array may be of any data type
+func FindMajority(input []interface{}) (result interface{}) {
```
Comment (erocs) [inline challenge_3/go/makernaren/max_occurance.go]: A slice of empty interfaces is unwieldy. Sure, it's Go's iffy "generics", but if you have a typed slice, it has to be copied into this new generic slice before being passed in. You would likely end up just writing a FindMajorityTYPE for each type you wanted this functionality for or you'd create a new interface that better supported what this needs to work.

For this problem, it's ok, as you create the slice of empty interfaces as input and it's not used for anything else.

File: challenge_3/go/makernaren/max_occurance.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
+// Take an array of mixed data types and find the element that occurs
+// most number of times.
+package main
```
Comment (erocs) [inline challenge_3/go/makernaren/max_occurance.go]: I would put the challenge functionality and unit tests in a challenge specific package and break out main() into its own file / package.

File: challenge_3/go/makernaren/max_occurance.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+//as elements in the given array may be of any data type
+func FindMajority(input []interface{}) (result interface{}) {
+	// Create a hashmap, where element of that array is key and its number of
+	// occurances is the value.
+	occurances := make(map[interface{}]int)
+	for _, key := range input {
+		occurances[key]++
+	}
+	// Return when we find the max occurance.
+	max := 0
```
Comment (erocs) [inline challenge_3/go/makernaren/max_occurance.go]: You can calculate this on the fly in the loop where you calculate occurrences. Just track what the current item / max count is and update that if the element you just incremented is larger.

File: challenge_3/go/makernaren/max_occurance.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	kv_pair := make([]interface{}, 2)
```
Comment (erocs) [inline challenge_3/go/makernaren/max_occurance.go]: Using a pair to track is odd. Just create two separate local variables each with their own descriptive name.

**Pull request merged**